pyhf/tensor/backend.py

```diff
@@ -0,0 +1,8 @@
+class Backend(object):
+    """General backend class for all of our backends"""
+
+    def __eq__(self, other):
+        if isinstance(other, Backend):
+            return self.__class__ == other.__class__
+        return False
```

Comment:
maybe  we should make this a string comparison with strings that are not `__class__`? e.g. I'm anticipating setting backends in the `pyhf` CLI by name `pyhf cls --backend numpy` and there we might not want to be tied to the class names

Reply:
done